### ISSUE TITLE
[Sidenav] Add optional(?) affordance for collapsible sections

### ISSUE THREAD

**Issue by castastrophe** [adobe/spectrum-css#1523]
## Description

It can be difficult as a consumer to recognize that a section is expandable when they are loaded in a collapsed state because there is no open/close affordance.

## Why do you need this feature or component?

On the [Spectrum Web Components](https://opensource.adobe.com/spectrum-web-components/) site, this component is used for the side navigation.  We have heard feedback from consumers that some of the sections were difficult to discover because the collapsible regions were hard to differentiate from the external links.

## Additional context

<img width="242" alt="Screen Shot 2022-09-27 at 1 09 52 PM" src="https://user-images.githubusercontent.com/1840295/192591757-a1431940-6dc1-47f7-8f2f-39440b86863c.png">

**Comment by jenndiaz** [comment]
I touched based with Design on 5/22/2023, they have previously discussed this and their recommendation is to "use Tree view instead for products that think the additional indication is needed! especially for SWC’s example where they’re mixing and matching the types"

**Comment by castastrophe** [comment]
I did try to use the treeview but it really doesn't work for side navigation. The indentation is too extreme and it lacks many of the utility classes needed to indicate state on a navigation.
